src/service/vaults.ts

```diff
@@ -1,60 +1,46 @@
 import { fetchRPCAddr, fetchVstorageKeys } from 'utils/rpc';
 import {
   keyForVault,
-  PriceDescription,
   useVaultStore,
   VaultInfoChainData,
   VaultManager,
   VaultMetrics,
 } from 'store/vaults';
 import { makeFollower, iterateLatest } from '@agoric/casting';
 import { appStore } from 'store/app';
-import type { BrandInfo } from 'store/app';
-import type { Brand, Amount } from '@agoric/ertp/src/types';
-import type { Ratio } from 'store/vaults';
 import { toast } from 'react-toastify';
 import { CapData } from '@endo/marshal';
-
-const issuerPetnameToPriceFeed = new Map<string, string>([
-  ['IbcATOM', 'ATOM-USD_price_feed'],
-]);
+import type { Brand, Amount } from '@agoric/ertp/src/types';
+import type { Ratio, PriceQuote } from 'store/vaults';
 
 type ValuePossessor<T> = {
   value: T;
 };
 
-type PriceFeedUpdate = ValuePossessor<PriceDescription>;
+type PriceFeedUpdate = ValuePossessor<{
+  quoteAmount: PriceQuote;
+  quotePayment: unknown;
+}>;
 
 // Subscribes to price feeds for new brands.
 const watchPriceFeeds = () => {
   let isStopped = false;
   const { leader, importContext } = appStore.getState();
 
-  const brandsToWatch = new Set<Brand>();
-  const watchedBrands = new Set<Brand>();
-  let brandToInfo: Map<Brand, BrandInfo> | null = null;
+  // Map of brands to any manager that publishes a price quote of that brand.
+  // If two managers have the same brand, their price quotes are the same, so
+  // we only need to watch one.
+  const brandsToWatch = new Map<Brand, string>();
```

Comment:
price quotes are for a tuple, in:BrandA and out:BrandB. So I think we need the key to be those tuples. Those are safe to deduplicate but not a single brand.

Reply:
Are we going to get vault price feeds where brand out is not IST? Tuples would be messy since we have to serialize them as strings.

Reply:
> Are we going to get vault price feeds where brand out is not IST?

Yes, I believe that's on the roadmap. Though it's probably far enough out to disregard for me.

> Tuples would be messy since we have to serialize them as strings.

Ok, let's leave it as is. But please clarify that we're tracking _collateral_ brands and assuming the debt brand is IST.

Reply:
Done